linera-storage-service/src/common.rs

```diff
@@ -16,18 +18,39 @@ const TEST_SHARED_STORE_MAX_CONCURRENT_QUERIES: usize = 10;
 const TEST_SHARED_STORE_MAX_STREAM_QUERIES: usize = 10;
 
 #[derive(Debug, Error)]
-pub enum SharedContextError {
+pub enum ServiceContextError {
     /// Not matching entry
     #[error("Not matching entry")]
     NotMatchingEntry,
 
+    /// Failed to find the storage_service_server binary
+    #[error("Failed to find the storage_service_server binary")]
+    FailedToFindStorageServiceServerBinary,
```

Comment:
Should this also use `#[from]` with whatever error `resolve_binary` returns?

Reply:
I think the right solution is to remove the `anyhow` from the `resolve_binary`.
But maybe, we can do that in a subsequent PR.